pkg/convert/pot.go

```diff
@@ -236,271 +234,98 @@ func textWithPlaceholders(mfBuilder *mf2.Builder, msg string, placeholders map[s
 
 // ---------------------------------------Translation->PO---------------------------------------
 
-type poTag string
-
-const (
-	MsgID        poTag = "msgid"
-	PluralID     poTag = "msgid_plural"
-	MsgStrPlural poTag = "msgstr[%d]"
-	MsgStr       poTag = "msgstr"
-)
-
-// ToPot function takes a model.Translation structure,
-// writes the language information and each message to a buffer in the POT file format,
-// and returns the buffer contents as a byte slice representing the POT file.
-// TODO instead of creating a buffer, create PO structure here and fill it with data.
-// Then convert PO structure to bytes.
-func ToPot(t model.Translation) ([]byte, error) {
-	var b bytes.Buffer
-
-	if _, err := fmt.Fprintf(&b, "msgid \"\"\nmsgstr \"\"\n\"Language: %s\\n\"\n", t.Language); err != nil {
-		return nil, fmt.Errorf("write language: %w", err)
+// ToPo converts a model.Translation structure to a byte slice representing a PO file.
+func ToPo(t model.Translation) ([]byte, error) {
+	po := pot.Po{
+		Header:   pot.HeaderNode{Language: t.Language},
+		Messages: make([]pot.MessageNode, 0, len(t.Messages)),
 	}
 
-	if !t.Original {
-		// Temporary we support plural forms (one and other).
-		// https://www.gnu.org/software/gettext/manual/html_node/Plural-forms.html
-		if _, err := fmt.Fprintf(&b, "\"Plural-Forms: nplurals=%d; plural=(n != 1);\\n\"\n\n", pluralCountLimit); err != nil {
-			return nil, fmt.Errorf("write Plural-Forms: %w", err)
-		}
-	}
+	var placeholders map[ast.Variable]string // MF2Variable:OriginalVariable, only for complex messages
 
-	for i, message := range t.Messages {
-		if err := writeMessage(&b, i, message); err != nil {
-			return nil, fmt.Errorf("write message: %w", err)
-		}
-	}
+	// patternsToMsg function converts a slice of patterns to a PO msgstr.
+	patternsToMsg := func(patterns []ast.Pattern) string {
+		var text string
 
-	return b.Bytes(), nil
-}
+		for _, p := range patterns {
+			switch p := p.(type) {
+			case ast.TextPattern:
+				text += string(p)
+			case ast.PlaceholderPattern:
+				//nolint:forcetypeassert // No other types are possible.
+				text += placeholders[p.Expression.(ast.VariableExpression).Variable]
+			}
+		}
 
-// writeToPoTag function selects the appropriate write function (writeDefault or writePlural)
-// based on the tag type (poTag) and uses that function to write the tag value to a bytes.Buffer.
-func writeToPoTag(b *bytes.Buffer, tag poTag, str string) error {
-	write := writeDefault
-	if tag == MsgStrPlural {
-		write = writePlural
+		return text
 	}
 
-	if err := write(b, tag, str); err != nil {
-		return fmt.Errorf("write: %w", err)
+	// If original, msgstr are empty
+	if t.Original {
+		patternsToMsg = func(patterns []ast.Pattern) string { return "" }
 	}
 
-	return nil
-}
+	unquoteLiteral := func(l ast.Literal) string { return strings.ReplaceAll(l.String(), "|", "") }
```

Comment:
We'll need to resolve how we handle raw and MS2 formatted values. Not part of this PR.

Reply:
created issue so we wont forget